### PR TITLE
feat(router-generator): prepend and append user content to the generated route-tree file

### PR DESCRIPTION
**Pull request by SeanCassiere**
From this comment: https://github.com/TanStack/router/pull/1496#issuecomment-2067631665

This change allows a user to prepend and append lines to the generated route-tree file by letting them configure an array of strings for each property.

### CODE REVIEW

File: packages/router-generator/src/config.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ export const configSchema = z.object({
   disableTypes: z.boolean().optional().default(false),
   addExtensions: z.boolean().optional().default(false),
   disableLogging: z.boolean().optional().default(false),
+  prependToRouteTreeFile: z.array(z.string()).optional().default([]),
```
Comment (SeanCassiere) [inline packages/router-generator/src/config.ts:16]: Let me know if you've got a better name for this configuration field.

File: packages/router-generator/src/generator.ts [hunk 2]
```diff
@@ -462,6 +462,7 @@ export async function generator(config: Config) {
     '// @ts-nocheck',
     '// noinspection JSUnusedGlobalSymbols',
     '// This file is auto-generated by TanStack Router',
+    ...config.prependToRouteTreeFile,
```
Comment (SeanCassiere) [inline packages/router-generator/src/generator.ts:465]: Do we want this to actually be prepended to the very top? or should it come after the stuff we are shipping in the file?

File: packages/router-generator/src/config.ts (hunk 1 above)
Comment (schiller-manuel) [reply packages/router-generator/src/config.ts:16]: how about `routeTreeFileHeader` ?

File: packages/router-generator/src/generator.ts [hunk 3]
```diff
@@ -462,6 +462,7 @@ export async function generator(config: Config) {
     '// @ts-nocheck',
     '// noinspection JSUnusedGlobalSymbols',
     '// This file is auto-generated by TanStack Router',
```
Comment (schiller-manuel) [inline packages/router-generator/src/generator.ts:464]: I would put all this into the default value of `routeTreeFileHeader`

File: packages/router-generator/src/config.ts [hunk 4]
```diff
@@ -13,6 +13,8 @@ export const configSchema = z.object({
...
+  appendToRouteTreeFile: z.array(z.string()).optional().default([]),
```
Comment (schiller-manuel) [inline packages/router-generator/src/config.ts:17]: how about `routeTreeFileFooter`?

File: packages/router-generator/src/generator.ts [hunk 5]
```diff
@@ -603,6 +604,7 @@ export async function generator(config: Config) {
         ]),
     '// Create and export the route tree',
     `export const routeTree = rootRoute.addChildren([${routeConfigChildrenText}])`,
+    ...config.appendToRouteTreeFile,
     '/* prettier-ignore-end */',
```
Comment (schiller-manuel) [inline packages/router-generator/src/generator.ts:608]: put `/* prettier-ignore-end */` into the default value of `routeTreeFileFooter`

File: packages/router-generator/src/generator.ts (hunk 3 above)
Comment (SeanCassiere) [reply packages/router-generator/src/generator.ts:464]: Will do, although I think its worth keeping this one outside the default list and something we ship with the generator always instead: // This file is auto-generated by TanStack Router

**Review by schiller-manuel — APPROVED** [review]

**Pull request merged**